src/main/java/com/denizenscript/ddiscordbot/commands/DiscordInteractionCommand.java

```diff
@@ -0,0 +1,566 @@
+                            }                            
+                            bestMatch.editCommand().setDefaultEnabled(isEnabled).complete();
+                            group.guild.updateCommandPrivilegesById(bestMatch.getIdLong(), privileges).complete();
+                            break;
+                        }
+                        case DELETE: {
+                            Command bestMatch = matchCommandByName(scriptEntry, name, client, group);
+                            if (bestMatch == null) {
+                                return;
+                            }
+                            if (group == null) {
+                                client.deleteCommandById(bestMatch.getIdLong()).complete();
+                            } 
+                            else {
+                                group.getGuild().deleteCommandById(bestMatch.getIdLong()).complete();
+                            }
+                            break;
+                        }
+                    }
+                    break;
+                }
+                case DEFER: {
+                    if (interaction == null) {
+                        Debug.echoError(scriptEntry.getContext(), "Must specify an interaction!");
+                        scriptEntry.setFinished(true);
+                        return;
+                    } 
+                    else if (interaction.getInteraction() == null) {
+                        Debug.echoError(scriptEntry.getContext(), "Invalid interaction! Has it expired?");
+                        scriptEntry.setFinished(true);
+                        return;
+                    }
+                    interaction.getInteraction().deferReply(isEphermal).complete();
+                    break;
+                }
+                case REPLY: {
+                    if (interaction == null) {
+                        Debug.echoError(scriptEntry.getContext(), "Must specify an interaction!");
+                        scriptEntry.setFinished(true);
+                        return;
+                    } 
+                    else if (interaction.getInteraction() == null) {
+                        Debug.echoError(scriptEntry.getContext(), "Invalid interaction! Has it expired?");
+                        scriptEntry.setFinished(true);
+                        return;
+                    }
+                    /**
+                     * Messages aren't allowed to have attachments in ephermal messages
+                     * Since you can't see if the acknowledged message is ephermal or not, this is a requirement so we don't have to try/catch
+                     */
+                    else if (message == null) {
+                        Debug.echoError(scriptEntry.getContext(), "Must have a message!");
+                        scriptEntry.setFinished(true);
+                        return;
+                    }
+
+                    // The following code is painful.
```

Comment:
This personal remark belongs in a Discord message with a few sad emoji or something, not published code